paddle/phi/kernels/funcs/layer_norm_impl.cu.h

```diff
@@ -295,6 +295,7 @@ __global__ __launch_bounds__(THREADS_PER_CTA) void fast_ln_fwd_kernel(
     }
 
     if (WARPS_N > 1) {
+      __syncthreads();
```

Comment:
Is this sync needed? or line 302 is enough?

Reply:
It is necessary. Without sync, some threads are reading smem at L274, and some are writing smem at 300 in parallel.